access-grant/src/main/java/com/inrupt/client/accessgrant/AccessGrantClient.java

```diff
@@ -181,6 +182,41 @@ public CompletionStage<AccessGrant> issue(final URI type, final URI agent, final
         });
     }
 
+    /**
+     * Verify an access grant or request.
+     *
+     * @param accessGrant the access grant to verify
+     * @param options the options to which this credential applies
+     * @return the next stage of completion containing the resulting credential
+     */
+    public CompletionStage<VerificationResponse> verify(final AccessGrant accessGrant, final Set<String> options) {
+        return v1Metadata().thenCompose(metadata -> {
+            
+            final Map<String, Object> presentation = new HashMap<>();
+            presentation.put(VERIFIABLE_CREDENTIAL, accessGrant);
+            presentation.put(OPTIONS, Arrays.asList(options));
+
+            final Request req = Request.newBuilder(metadata.verifyEndpoint)
+                .header(CONTENT_TYPE, APPLICATION_JSON)
+                .POST(Request.BodyPublishers.ofByteArray(serialize(presentation))).build();
+
+            return client.send(req, Response.BodyHandlers.ofInputStream())
+                .thenApply(res -> {
+                    try (final InputStream input = res.body()) {
+                        final int status = res.statusCode();
+                        if (isSuccess(status)) {
+                            return processVerificationResult(input);
+                        }
+                        throw new AccessGrantException("Unable to perform Access Grant verify: HTTP error " + status,
```

Comment:
In the case of an error the response will also contain an `error`, `errorDescription` and an optional `reason`. These could be useful for a downstream client to know but they are getting dropped here and we are only giving them the `status`.

Reply:
According to the VC-API spec, the verify endpoint will return a 200 in cases where the credential is invalid, but there is no defined response for, say, 4xx or 5xx category errors.

I think it would be great for the client to _try_ to parse 4xx/5xx error bodies as if they contain those response, but that can be part of an overall improvement in how errors are processed.